theseus/geometry/se3.py

```diff
@@ -232,35 +234,103 @@ def exp_map(tangent_vector: torch.Tensor) -> "SE3":
+                near_zero, torch.zeros_like(theta), theta_minus_sine_by_theta3_t
+            )
+            jac = torch.zeros(
+                tangent_vector.shape[0],
+                6,
+                6,
+                dtype=tangent_vector.dtype,
+                device=tangent_vector.device,
+            )
+            jac[:, :3, :3] = (
+                theta_minus_sine_by_theta3_rot
+                * tangent_vector_ang.view(-1, 3, 1)
+                @ tangent_vector_ang.view(-1, 1, 3)
+            )
+            diag_jac = jac.diagonal(dim1=1, dim2=2)
+            diag_jac += sine_by_theta.view(-1, 1)
+
+            jac_temp_rot = one_minus_cosine_by_theta2.view(
+                -1, 1
+            ) * tangent_vector_ang.view(-1, 3)
+
+            jac[:, 0, 1] += jac_temp_rot[:, 2]
+            jac[:, 1, 0] -= jac_temp_rot[:, 2]
+            jac[:, 0, 2] -= jac_temp_rot[:, 1]
+            jac[:, 2, 0] += jac_temp_rot[:, 1]
+            jac[:, 1, 2] += jac_temp_rot[:, 0]
+            jac[:, 2, 1] -= jac_temp_rot[:, 0]
+
+            jac[:, 3:, 3:] = jac[:, :3, :3]
+
+            d_one_minus_cosine_by_theta2 = torch.where(
+                near_zero,
+                -1 / 12.0,
+                (sine_by_theta - 2 * one_minus_cosine_by_theta2) / theta2_nz,
+            )
+            d_theta_minus_sine_by_theta3 = torch.where(
+                near_zero,
+                -1 / 60.0,
+                (one_minus_cosine_by_theta2 - 3 * theta_minus_sine_by_theta3_t)
+                / theta2_nz,
+            )
+
+            w = tangent_vector[:, 3:]
+            v = tangent_vector[:, :3]
+            wv = w.cross(v)
+            wwv = w.cross(wv)
+            sw = theta_minus_sine_by_theta3_t.view(-1, 1) * w
+
+            jac_temp_t = (
+                d_one_minus_cosine_by_theta2.view(-1, 1) * wv
+                + d_theta_minus_sine_by_theta3.view(-1, 1) * wwv
+            ).view(-1, 3, 1) @ w.view(-1, 1, 3)
+            jac_temp_t -= v.view(-1, 3, 1) @ sw.view(-1, 1, 3)
+            jac_temp_v = (
+                -one_minus_cosine_by_theta2.view(-1, 1) * v
+                - theta_minus_sine_by_theta3_t.view(-1, 1) * wv
+            )
+            jac_temp_t += SO3.hat(jac_temp_v)
+            diag_jac_t = torch.diagonal(jac_temp_t, dim1=1, dim2=2)
+            diag_jac_t += (sw.view(-1, 1, 3) @ v.view(-1, 3, 1)).view(-1, 1)
+
+            jac[:, :3, 3:] = ret[:, :, :3].transpose(1, 2) @ jac_temp_t
+
+            jacobians.append(jac)
+
         return ret
 
     def _log_map_impl(self) -> torch.Tensor:
         NEAR_PI_EPS = 1e-7
         NEAR_ZERO_EPS = 5e-3
 
-        ret = torch.zeros(self.shape[0], 6, dtype=self.dtype, device=self.device)
-
-        ret[:, 3] = 0.5 * (self[:, 2, 1] - self[:, 1, 2])
-        ret[:, 4] = 0.5 * (self[:, 0, 2] - self[:, 2, 0])
-        ret[:, 5] = 0.5 * (self[:, 1, 0] - self[:, 0, 1])
+        sine_axis = torch.zeros(self.shape[0], 3, dtype=self.dtype, device=self.device)
```

Comment:
This part was already merged, correct? Is this GitHub not syncing correctly?

Reply:
Yes.